sources/api/pluto/src/main.rs

```diff
@@ -327,60 +310,54 @@ async fn get_private_dns_name(client: &mut ImdsClient) -> Result<String> {
-    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
-    eprintln!(
-        r"Usage: {} [max-pods | cluster-dns-ip | node-ip | provider-id | private-dns-name]",
-        program_name
-    );
-    process::exit(1);
-}
-
-/// Parses args for the setting key name.
-fn parse_args(mut args: env::Args) -> String {
-    args.nth(1).unwrap_or_else(|| usage())
-}
-
 async fn run() -> Result<()> {
-    let setting_name = parse_args(env::args());
     let mut client = ImdsClient::new();
+    let aws_k8s_info = api::get_aws_k8s_info().await.context(error::AwsInfoSnafu)?;
 
-    let setting = match setting_name.as_ref() {
-        "cluster-dns-ip" => get_cluster_dns_ip(&mut client).await,
-        "node-ip" => get_node_ip(&mut client).await,
-        // If we want to specify a reasonable default in a template, we can exit 2 to tell
-        // sundog to skip this setting.
-        "max-pods" => get_max_pods(&mut client)
-            .await
-            .map_err(|_| process::exit(2)),
-        "provider-id" => get_provider_id(&mut client).await,
-        "private-dns-name" => get_private_dns_name(&mut client).await,
-        _ => usage(),
-    }?;
-
-    // sundog expects JSON-serialized output so that many types can be represented, allowing the
-    // API model to use more accurate types.
-
-    // 'max_pods' setting is an unsigned integer, convert 'settings' to u32 before serializing to JSON
-    if setting_name == "max-pods" {
-        let max_pods = serde_json::to_string(
-            &setting
-                .parse::<u32>()
-                .context(error::ParseToU32Snafu { setting: &setting })?,
-        )
-        .context(error::OutputJsonSnafu { output: &setting })?;
-        println!("{}", max_pods);
+    // Generate the cluster-dns-ip
+    let cluster_dns_ip = get_cluster_dns_ip(&mut client, &aws_k8s_info).await?;
+    // Generate the node-ip
+    let node_ip = get_node_ip(
+        &mut client,
+        IpAddr::from_str(&cluster_dns_ip).context(error::BadIpSnafu {
+            ip: cluster_dns_ip.clone(),
+        })?,
+    )
+    .await?;
+
+    // Generate max-pods
+    // Skip the set if this fails
+    let max_pods = if let Ok(max_pods) = get_max_pods(&mut client).await {
+        Some(max_pods)
     } else {
-        let output =
-            serde_json::to_string(&setting).context(error::OutputJsonSnafu { output: &setting })?;
-        println!("{}", output);
-    }
+        None
+    };
+
+    // Generate provider-id
+    let provider_id = get_provider_id(&mut client).await?;
+
+    // Generate private-dns-name
+    let private_dns_name = get_private_dns_name(&mut client, &aws_k8s_info).await?;
+
+    let generated_settings: serde_json::Value = serde_json::json!({
+        "kubernetes": {
+            "node-ip": node_ip,
+            "cluster-dns-ip": cluster_dns_ip,
+            "max-pods": max_pods,
+            "provider-id": provider_id,
+            "hostname-override": private_dns_name
```

Comment:
[KubernetesSettings](https://github.com/bottlerocket-os/bottlerocket/blob/64eff582e8738ce9e4dbf21f485791b225bac319/sources/models/src/lib.rs#L316) says:
```
// Generated in `aws-k8s-1.26*` variants only
```

This is a problem the new `pluto` will have to handle somehow. At the time in #3033, we felt the risk of a breaking behavior change from generating this as a new default was too great, so it only applied to 1.26 and beyond.

One option would be a `pluto.toml` that we can package in the different Kubernetes packages, but that's going to be weird since they're not currently platform-specific.

Another would be to pull the variant ID out of `apiclient get os.variant_id` and see if it matches the problematic prefixes - `aws-k8s-1.23` through `aws-k8s-1.25`, and if so don't include a hostname override.